lib/index.js

```diff
@@ -179,6 +180,9 @@ Segment.prototype.initialize = function() {
     this.cookie('segment_cross_domain_id_timestamp', null);
   }
 
+  // Delete cross domain identifiers.
+  this.deleteCrossDomainId();
```

Comment:
a bit nitpicking, but I find the function name confusing. Here it looks like we're going to always delete the cross domain id, which is not the case. Maybe _checkCrossDomainIdForDeletion()_, or something better?

Reply:
updating to `deleteCrossDomainIdIfNeeded()`